bedrock/security/views.py

```diff
@@ -205,19 +206,22 @@ class KVRedirectsView(CachedRedirectView):
     prod_ver_re = re.compile('(\w+)(\d{2})$')
 
     def get_redirect_url(self, *args, **kwargs):
-        url_component = kwargs['filename']
-        if url_component == 'suite17':
-            return reverse('security.product-advisories', kwargs={'slug': 'mozilla-suite'})
-
-        match = self.prod_ver_re.match(url_component)
-        if match:
-            product, version = match.groups()
-            version = '{0}.{1}'.format(*version)
-            return reverse('security.product-version-advisories', kwargs={'product': product,
-                                                                          'version': version})
-
-        if url_component.endswith('ESR'):
-            return reverse('security.product-advisories',
-                           kwargs={'slug': url_component[:-3] + '-esr'})
-
-        return reverse('security.product-advisories', kwargs={'slug': url_component})
+        try:
+            url_component = kwargs['filename'].replace(' ', '')
+            if url_component == 'suite17':
+                return reverse('security.product-advisories', kwargs={'slug': 'mozilla-suite'})
+
+            match = self.prod_ver_re.match(url_component)
+            if match:
+                product, version = match.groups()
+                version = '{0}.{1}'.format(*version)
+                return reverse('security.product-version-advisories', kwargs={'product': product,
+                                                                              'version': version})
+
+            if url_component.endswith('ESR'):
+                return reverse('security.product-advisories',
+                               kwargs={'slug': url_component[:-3] + '-esr'})
+
+            return reverse('security.product-advisories', kwargs={'slug': url_component})
+        except NoReverseMatch:
```

Comment:
NIT: there is a non-trivial amount of logic inside this try block that has nothing to do with the actual exception. It works, but what if instead of having three different `return reverse(...)` that could each raise a `NoReverseMatch` you just stored the `viewname` and `reverse_kwargs` for each of them in variables and had a single call to `return reverse(viewname, reverse_kwargs)` inside the try block?